### ISSUE TITLE
Wish: gulp-real-favicon without requests to http://realfavicongenerator.net/

### ISSUE THREAD

**Issue by HarryBurns** [RealFaviconGenerator/gulp-real-favicon#4]
Hello there!
This is a feature request.

I want to use your great tool in continuous integration cycle.
My production build server is located in isolated network (because of security policy), so I can't reach http://realfavicongenerator.net for icon generation from there.

Could you please make local version of real-favicon that will generate icons without requests to any web services?
I think, configs for your tool still can be generated on your site to provide money support through ads.

**Comment by phbernard** [comment]
TL; DR @haydenbleasel and I are working on it but that won't be ready any soon.

An offline version of RFG? @haydenbleasel and I are working on https://github.com/haydenbleasel/favicons . This module can already generate icons locally, but it doesn't take all RFG options into account (margins, white silhouette, etc.).

The first step is to make gulp-real-favicon use this module, instead of rfg-api (which is common to the Grunt and Gulp plugins). This requires some work, but it's not that complex.

The second step will be for favicons to support the same set of options RFG does. This is a very challenging task: there are a lot of options, plus favicons is a pure JS module while RFG uses external tools such as ImageMagick, Potrace, etc.

**Comment by haydenbleasel** [comment]
@HarryBurns What Philippe said. I'd recommend using Favicons for now as we work towards replicating RFG, or if you need specific functions that only RFG has at the moment, then raise an issue on the Favicons repo.

**Comment by HarryBurns** [comment]
Ok, thanks! I'll move all my wishes into [favicons](https://github.com/haydenbleasel/favicons) repo :)

By the way. Will you guys fix `gulp-real-favicon` to work with [favicons](https://github.com/haydenbleasel/favicons) or you will create new plugin?

**Comment by haydenbleasel** [comment]
@HarryBurns Sounds good. Not sure what's happening with `gulp-real-favicon`. I only plan on handling `favicons` and it's official wrappers. RFG is free to utilise these modules if needed.
